shared/modules/Numeric.ts

```diff
@@ -0,0 +1,620 @@
+    return new Numeric(
+      this.value.add(alignOperandDenominations(this, numeric).value),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  /**
+   * TODO: make it possible to subtract ETH - GWEI value. So if you have
+   * Numeric 1 with denomination ETH and Numeric 2 with Denomination WEI,
+   * first convert Numeric 2 to ETH then subtract the amount from Numeric 1.
+   *
+   * @param value
+   * @param base
+   * @param denomination
+   */
+  minus(
+    value: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    const numeric = Numeric.from(value, base, denomination);
+
+    return new Numeric(
+      this.value.minus(alignOperandDenominations(this, numeric).value),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  times(
+    multiplier: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    const multiplierNumeric = Numeric.from(multiplier, base, denomination);
+    return new Numeric(
+      this.value.times(
+        alignOperandDenominations(this, multiplierNumeric).value,
+      ),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  /**
+   * Divides the Numeric by another supplied Numeric, carrying over the base
+   * and denomination from the current Numeric.
+   *
+   * @param divisor - The Numeric to divide this Numeric by
+   * @param base
+   * @param denomination
+   * @returns A new Numeric that contains the result of the division
+   */
+  divide(
+    divisor: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return new Numeric(
+      this.value.div(
+        alignOperandDenominations(
+          this,
+          Numeric.from(divisor, base, denomination),
+        ).value,
+      ),
+      this.base,
+      this.denomination,
+    );
+  }
+
+  greaterThan(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.greaterThan(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  greaterThanOrEqualTo(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.greaterThanOrEqualTo(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  lessThan(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.lessThan(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  lessThanOrEqualTo(
+    comparator: Numeric | NumericValue,
+    base?: NumericBase,
+    denomination?: EtherDenomination,
+  ) {
+    return this.value.lessThanOrEqualTo(
+      Numeric.from(comparator, base, denomination).value,
+    );
+  }
+
+  isNegative() {
+    return this.value.isNegative();
+  }
+
+  isPositive() {
+    return this.isNegative() === false;
```

Comment:
just following the conversation here:
https://github.com/MetaMask/metamask-extension/pull/17262#discussion_r1082646325  

Should we add 
```
isZero() {}
```
to differ ? 

or we can put 
```
isNegative() {
    return this.value.isNegative() && this.value !== 0;
  }


  isPositive() {
    return this.isNegative() === false && this.value !== 0;
}
```

Or we can do 
```
isPositive() {
    return this.value.isPositive();
    }
```

wdyt?

Reply:
There are no current parts of the codebase that need a toZero. 0 is positive so returning true from isPositive is correct for 0.  We can add a toZero method in the future if it is needed.

Reply:
we primarily use isNegative to determine if someone entered a negative number in the send screen, but you can send 0 eth.

Reply:
Oki gotcha !

Reply:
I'd like to rephrase my comment from 'there are no current parts of the codebase that need a toZero' to 'there are no known (by me) instances of BigNumber comparisons that need  the .isZero method'. BigNumber does expose a isZero method so there is definitely a place for this method on Numeric in the future. We also *do* compare string '0's elsewhere that if we end up using Numeric in more places might result in a need for a 0 check. :)